src/Honeycomb.Samplers/DeterministicSampler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+using OpenTelemetry.Trace;
+
+namespace Honeycomb.Samplers
+{
+    public class DeterministicSampler : Sampler, IDisposable
+    {
+        private const string DescriptionFormat = "DeterministicSampler({0})";
+        private const string SampleRateAttributeName = "sampleRate";
+        private const string Hyphen = "-";
+        private const int Zero = 0;
+        private const int Four = 4;
+        private const int One = 1;
+        private const int Base16 = 16;
+        private readonly SHA1 sha1 = SHA1.Create();
+        private readonly int sampleRate;
+        private readonly long upperBound;
+
+        public DeterministicSampler(int sampleRate)
+        {
+            if (sampleRate < Zero)
+            {
+                throw new ArgumentOutOfRangeException("Sample rate must not be negative.");
+            }
+
+            this.sampleRate = sampleRate;
+            this.Description = string.Format(DescriptionFormat, this.sampleRate.ToString(CultureInfo.InvariantCulture));
+            this.upperBound = sampleRate == Zero ? Zero : uint.MaxValue / sampleRate;
+        }
+
+        public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
+        {
+            if (sampleRate == One)
+            {
+                return CreateResult(SamplingDecision.RecordAndSample, One);
+            }
+            if (sampleRate == Zero)
+            {
+                return CreateResult(SamplingDecision.Drop, Zero);
+            }
+
+            var bytes = Encoding.UTF8.GetBytes(samplingParameters.TraceId.ToString());
+            var hash = sha1.ComputeHash(bytes);

```

Comment:
not sure what the cost of instantiating the SHA1 object is but it might be better to just use a `using` statement here

Reply:
That was the trade-off we discussed in this.  Mike moved it out of the method for that reason.  Disposing it as part of the object's dispose method would need to be done though

Reply:
yeh, dispose is a pain in the butt to implement properly though and we would need to do more than is in this PR right now

Reply:
I did originally use a `using` statement, but as @martinjt mentioned, we discussed and moved to a readonly field and dispose during object close. See below for the dispose pattern usage below.